src/esp32/single_encoded_motor.rs

```diff
@@ -0,0 +1,78 @@
+use super::single_encoder::SingleEncoderType;
+use crate::common::encoder::{
+    Direction, Encoder, EncoderPositionType, EncoderSupportedRepresentations, SingleEncoder,
+};
+use crate::common::motor::{Motor, MotorType};
+
+use crate::common::status::Status;
+use std::collections::BTreeMap;
+
+pub struct SingleEncodedMotor {
+    encoder: SingleEncoderType,
+    motor: MotorType,
+}
+
+impl SingleEncodedMotor {
+    pub fn new(motor: MotorType, encoder: SingleEncoderType) -> Self {
+        Self { encoder, motor }
+    }
+}
+
+impl Motor for SingleEncodedMotor {
+    fn set_power(&mut self, power_pct: f64) -> anyhow::Result<()> {
+        let dir = match power_pct {
+            x if x > 0.0 => Direction::Forwards,
+            x if x < 0.0 => Direction::Backwards,
+            x if x == 0.0 => {
+                let prev_dir = self.encoder.get_direction()?;
+                match prev_dir {
+                    Direction::Backwards | Direction::StoppedBackwards => {
+                        Direction::StoppedBackwards
+                    }
+                    Direction::Forwards | Direction::StoppedForwards => Direction::StoppedForwards,
+                }
+            }
+            _ => anyhow::bail!("how did we get here?"),
```

Comment:
use unreachable!() or Ord::cmp :)

Reply:
thanks, I was wondering how I was supposed to deal with that